dev/run-tests.py

```diff
@@ -0,0 +1,536 @@
+                               "mllib/"])]
+        streaming_files = [f for f in changed_files
+                           if any(f.startswith(p) for p in
+                                  ["examples/scala-2.10/",
+                                   "examples/src/main/java/org/apache/spark/examples/streaming/",
+                                   "examples/src/main/scala/org/apache/spark/examples/streaming/",
+                                   "external/",
+                                   "extras/java8-tests/",
+                                   "extras/kinesis-asl/",
+                                   "streaming/"])]
+        graphx_files = [f for f in changed_files
+                        if any(f.startswith(p) for p in
+                               ["examples/src/main/scala/org/apache/spark/examples/graphx/",
+                                "graphx/"])]
+        doc_files = [f for f in changed_files if f.startswith("docs/")]
+
+        # union together all changed top level project files
+        top_level_project_files = set().union(*[set(f) for f in [sql_files,
+                                                                 mllib_files,
+                                                                 streaming_files,
+                                                                 graphx_files,
+                                                                 doc_files]])
+        changed_core_files = set(changed_files).difference(top_level_project_files)
+
+        if changed_core_files:
+            changed_modules.add("CORE")
+        if sql_files:
+            print "[info] Detected changes in SQL. Will run Hive test suite."
+            changed_modules.add("SQL")
+        if mllib_files:
+            print "[info] Detected changes in MLlib. Will run MLlib test suite."
+            changed_modules.add("MLLIB")
+        if streaming_files:
+            print "[info] Detected changes in Streaming. Will run Streaming test suite."
+            changed_modules.add("STREAMING")
+        if graphx_files:
+            print "[info] Detected changes in GraphX. Will run GraphX test suite."
+            changed_modules.add("GRAPHX")
+        if doc_files:
+            print "[info] Detected changes in documentation. Will build spark with documentation."
+            changed_modules.add("DOCS")
+
+        return changed_modules
+    else:
+        # we aren't in the Amplab environment so simply run all tests
+        changed_modules.add("ALL")
+        return changed_modules
+
+
+def run_scala_tests_maven(test_profiles):
+    mvn_test_goals = ["test", "--fail-at-end"]
+    profiles_and_goals = test_profiles + mvn_test_goals
+
+    print "[info] Running Spark tests using Maven with these arguments:",
+    print " ".join(profiles_and_goals)
+
+    exec_maven(profiles_and_goals)
+
+
+def run_scala_tests_sbt(test_modules, test_profiles):
+    # declare the variable for reference
+    sbt_test_goals = None
```

Comment:
Hey Josh, I'm out of pocket until tomorrow morning. Can you get a hot fix in for this? Immediately I'm not seeing what the issue is / code to fix it :/ I can figure it out tomorrow morning though if it can wait.



-----Original Message-----
From: Josh Rosen [notifications@github.com<mailto:notifications@github.com>]
Sent: Wednesday, June 17, 2015 09:58 PM Eastern Standard Time
To: apache/spark
Cc: York, Brennon
Subject: Re: [spark] [SPARK-7017][Build][Project Infra]: Refactor dev/run-tests into Python (#5694)


In dev/run-tests.py<https://github.com/apache/spark/pull/5694#discussion_r32693784>:


> +        return changed_modules
> +
> +
> +def run_scala_tests_maven(test_profiles):
> +    mvn_test_goals = ["test", "--fail-at-end"]
> +    profiles_and_goals = test_profiles + mvn_test_goals
> +
> +    print "[info] Running Spark tests using Maven with these arguments:",
> +    print " ".join(profiles_and_goals)
> +
> +    exec_maven(profiles_and_goals)
> +
> +
> +def run_scala_tests_sbt(test_modules, test_profiles):
> +    # declare the variable for reference
> +    sbt_test_goals = None


We'll end up concatenating a list to this if we only run the MLLib tests in the code below. This breaks PRB tests for MLLib, Streaming, and GraphX only PRs.

I'm considering reverting this patch or pushing a hotfix to address this.

—
Reply to this email directly or view it on GitHub<https://github.com/apache/spark/pull/5694/files#r32693784>.
________________________________________________________

The information contained in this e-mail is confidential and/or proprietary to Capital One and/or its affiliates and may only be used solely in performance of work or services for Capital One. The information transmitted herewith is intended only for use by the individual or entity to which it is addressed. If the reader of this message is not the intended recipient, you are hereby notified that any review, retransmission, dissemination, distribution, copying or other use of, or taking of any action in reliance upon this information is strictly prohibited. If you have received this communication in error, please contact the sender and delete the material from your computer.